### PR TITLE
Update onewire integration docs

### PR DESCRIPTION
**Pull request by MrDadoo**
Changes to go along with pull request #31419 for the onewire component.

## Proposed change

Changes made to describe platform better, how to use it, what it provides and what kind of devices it supports.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase:
https://github.com/home-assistant/home-assistant/pull/31419

- This PR fixes or closes issue:

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

**Comment by MrDadoo** [comment]
Am I supposed to do something else with these pull requests?
One of the tests seems not willing to complete.

**Comment by klaasnicolaas** [comment]
You must first resolve the merge conflict 😉

**Comment by MrDadoo** [comment]
Actually I don't understand why there was a conflict, but its resolved anyhow.

Can I in some way see who wrote the original docs... There are some things I dont really understand and would like to ask.

**Comment by klaasnicolaas** [comment]
I see that @fabaff originally made the documentation of this integration, but you can just ask your questions here. Maybe I can help you?

**Comment by MrDadoo** [comment]
There is a section about "multiple sensors" and "updating database using SQL"  and "discontinuity in measurements".  
I don't understand why this happens. Is it something related to the TAI-8570 device.

**Comment by MrDadoo** [comment]
One other question @klaasnicolaas. I am not up to date with new naming of home assistant.
I tried in the doc to discuss hass.io and Docker, but the linter didn't like that. 
What is the correct terminology for the softwar  package containing docker, supervisor, and homeassistnat that previously was called hass.io.
And the home assistant software that nowadays run within the docker container, what is that called?

**Comment by frenck** [comment]
Wat previously was called "Hass.io" is now called "Home Assistant".
"Home Assistant" is the full suite of projects we run.

Wat previously was called "Home Assistant" is now called "Home Assistant Core".
You can run Home Assistant Core in a Python virtual environment, or, using Docker.

There is also a variant "Hass.io on a generic 
Linux installation", in which people run the Supervisor on e.g., Ubuntu or Debian. This is now called "Home Assistant Supervised".

**Comment by MrDadoo** [comment]
Thanks allot for the clarification  @frenck. I'll keep that in mind. I'll use that when updating the docs

File: source/_integrations/onewire.markdown [hunk 1]
```diff
@@ -12,53 +12,56 @@ ha_codeowners:
 
 The `onewire` platform supports sensors which are using the One wire (1-wire) bus for communication.
 
-Supported devices:
+Each and every 1-wire device has a (globaly) unique ID that identifies the deivce on the bus. The first two digits identifies a device family and the last 14 digita are a globally unique number given to it during manufacturing.
```
Comment (frenck) [inline source/_integrations/onewire.markdown]:
```suggestion
Each and every 1-wire device has a (globally) unique ID that identifies the device on the bus. The first two digits identify a device family and the last 14 digits are a globally unique number given to it during manufacturing.
```

File: source/_integrations/onewire.markdown [hunk 2]
```diff
@@ -12,53 +12,56 @@ ha_codeowners:
...
+| 26     | [DS2438/B1-R1-A/S2-R1-A [2]](https://datasheets.maximintegrated.com/en/ds/DS2438.pdf)| Temperature, Voltage, Current, (pressure and humidity when using TAI-8570[1]) | 
+| 28     | [DS18B20](https://datasheets.maximintegrated.com/en/ds/DS18B20.pdf)          | Temperature                     |
+| 3B     | [DS1825](https://datasheets.maximintegrated.com/en/ds/DS1825.pdf)            | Temperature                     |
+| 42     | [DS28EA00](https://datasheets.maximintegrated.com/en/ds/DS28EA00.pdf)        | Temperature                     |
+| EF     | [HobbyBoard [3]](https://hobbyboards.com/)                                   | Temperature, Humidity, Moisture |  
 
-In order to setup 1-Wire support on Raspberry Pi, you'll need to edit `/boot/config.txt` following [this documentation](https://www.waveshare.com/wiki/Raspberry_Pi_Tutorial_Series:_1-Wire_DS18B20_Sensor#Enable_1-Wire).
-To edit `/boot/config.txt` on the Home Assistant Operating System, use [this documentation](https://developers.home-assistant.io/docs/en/hassio_debugging.html) to enable SSH and edit `/mnt/boot/config.txt` via `vi`.
+#### Notes:
```
Comment (frenck) [inline source/_integrations/onewire.markdown]:
```suggestion
## Notes

```

File: source/_integrations/onewire.markdown [hunk 3]
```diff
@@ -12,53 +12,56 @@ ha_codeowners:
...
+[1] The TAI-8570 Pressure Sensor is based on a 1-wire composite device by AAG Electronica. It contains, above 1-wire components, also barometer, hygrometer and illiminance sensors. This onewire platform can read and present values from that device.
```
Comment (frenck) [inline source/_integrations/onewire.markdown]:
```suggestion
- The TAI-8570 Pressure Sensor is based on a 1-wire composite device by AAG Electronica. It contains, above 1-wire components, also a barometer, hygrometer and illuminance sensors. This onewire platform can read and present values from that device.
```

File: source/_integrations/onewire.markdown [hunk 4]
```diff
@@ -12,53 +12,56 @@ ha_codeowners:
...
 
-## Interface adapter setup
+[2] Each 1-wire component data sheet describes the different properties the component provides. The owfs software adds some extra toos to make it easier for diy implementers to use the component. By hobbyists it is quite common to use DS2438 Smart Battery Monitor as a multipurpose measurement node that can place temperature, humidity and luminosity on the 1-wire bus by just adding some standard componentes to the DS2438. For different component types there is ready made algorithms implemented in owfs. Those are exposed by the owfs softare and can be red by this platform. The B1-R1-A/pressure is exposed as a barrometric pressure sensor. S2-R1-A/illuminance is presented as an illuminance sensor. For a more detailed description on these properties referr to the [owfs documentation](https://owfs.org/index_php_page_ds2438.html).
```
Comment (frenck) [inline source/_integrations/onewire.markdown]:
```suggestion
- Each 1-wire component data sheet describes the different properties the component provides. The owfs software adds some extra tools to make it easier for DIY implementers to use the component. By hobbyists, it is quite common to use DS2438 Smart Battery Monitor as a multipurpose measurement node that can place temperature, humidity and luminosity on the 1-wire bus by just adding some standard components to the DS2438. For different component types, there are ready-made algorithms implemented in owfs. Those are exposed by the owfs software and can be read by this platform. The B1-R1-A/pressure is exposed as a barometric pressure sensor. S2-R1-A/illuminance is presented as an illuminance sensor. For a more detailed description of these properties refer to the [owfs documentation](https://owfs.org/index_php_page_ds2438.html).
```

File: source/_integrations/onewire.markdown [hunk 5]
```diff
@@ -12,53 +12,56 @@ ha_codeowners:
...
+For this component the more basic quantities temperature, VAD, VDD and IAD is exported as separate sensors.
```
Comment (frenck) [inline source/_integrations/onewire.markdown]:
```suggestion
  For this component, the more basic quantities temperature, VAD, VDD and IAD is exported as separate sensors.
```

File: source/_integrations/onewire.markdown [hunk 6]
```diff
@@ -12,53 +12,56 @@ ha_codeowners:
...
 
-### owfs
+[3] Hobbyboards is a company that have been selling diy boards of diffent kinds. The company has been away from the market for some, time so no reference to the boards can be made. This platform have implementation for some of those.
```
Comment (frenck) [inline source/_integrations/onewire.markdown]:
```suggestion
- Hobbyboards is a company that has been selling DIY boards of different kinds. The company has been away from the market for some, time so no reference to the boards can be made. This platform has an implementation for some of those.
```

File: source/_integrations/onewire.markdown [hunk 7]
```diff
@@ -12,53 +12,56 @@ ha_codeowners:
...
 
-### owserver
+The 1-Wire bus can be connected directly to the IO pins of Raspberry Pi or by using a dedicated interface adapter, for example 
+[DS9490R](https://datasheets.maximintegrated.com/en/ds/DS9490-DS9490R.pdf) or adapters based on [DS2482-100](https://datasheets.maximintegrated.com/en/ds/DS2482-100.pdf) that can be directly attached to the IO pins on the Raspberry Pi.
 
-When an interface adapter is used, you can also access sensors on a remote or local Linux host that is running owserver.  owserver by default runs on port 4304. Use the `host` option to specify the host or IP of the remote server, and the optional `port` option to change the port from the default.
+It is also possible for this platform to interface with a remote 1-wire host over a network connection using ofws and owserver.
 
-### Units with multiple sensors
+## Raspberry Pi setup
```
Comment (frenck) [inline source/_integrations/onewire.markdown]:
```suggestion
## Raspberry Pi set up
```

File: source/_integrations/onewire.markdown [hunk 8]
```diff
@@ -12,53 +12,56 @@ ha_codeowners:
...
 
-This platform works with devices with multiple sensors which will cause a discontinuity in recorded values. Existing devices will receive a new ID and therefore show up as new devices.
-If you wish to maintain continuity it can be resolved in the database by renaming the old devices to the new names.
+In order to setup 1-Wire support on Raspberry Pi, you'll need to edit `/boot/config.txt` following [this documentation](https://www.waveshare.com/wiki/Raspberry_Pi_Tutorial_Series:_1-Wire_DS18B20_Sensor#Enable_1-Wire).
+To edit `/boot/config.txt` on the Home Assistant Operating System, use [this documentation](https://developers.home-assistant.io/docs/en/hassio_debugging.html) to enable SSH and edit `/mnt/boot/config.txt` via `vi`.
 
-Connect to your database using the instructions from [Database section](/docs/backend/database/). Check the names of sensors:
+When using the GPIO pins on Raspberry Pi directrly as a 1-wire bus, the description above uses two kernel modules. 1w_gpio that implements the 1-wire protocoll and 1w_therm that understands the DS18B20 (family 28) components inner structure and reports temperature.
```
Comment (frenck) [inline source/_integrations/onewire.markdown]:
```suggestion
When using the GPIO pins on Raspberry Pi directly as a 1-wire bus, the description above uses two kernel modules. `1w_gpio`, that implements the 1-wire protocol, and `1w_therm`, that understands the DS18B20 (family 28) components inner structure and reports temperature. 
```

File: source/_integrations/onewire.markdown [hunk 9]
```diff
@@ -12,53 +12,56 @@ ha_codeowners:
...
 
-Alter the names of sensors using the following examples:
+### owfs
 
-```sql
-UPDATE states SET entity_id='sensor.<sensor_name>_temperature' WHERE entity_id LIKE 'sensor.<sensor_name>%' AND attributes LIKE '%\u00b0C%';
-UPDATE states SET entity_id='sensor.<sensor_name>_pressure' WHERE entity_id LIKE 'sensor.<sensor_name>%' AND attributes LIKE '%mb%';
-UPDATE states SET entity_id='sensor.<sensor_name>_humidity' WHERE entity_id LIKE 'sensor.<sensor_name>%' AND attributes LIKE '%%%' ESCAPE '';
-```
+When an interface adapter is used, sensors can be accessed on Linux hosts via [owfs 1-Wire file system](https://owfs.org/). When using an interface adapter and the owfs, the `mount_dir` option must be configured to correspond a directory, where owfs device tree has been mounted. On systems where Home Assistant runs in a Docker container. `owfs` cannot escape that environment and hence cannot not populate the `mount_dir`. Use the owserver method on these systems instead.
```
Comment (frenck) [inline source/_integrations/onewire.markdown]:
```suggestion
When an interface adapter is used, sensors can be accessed on Linux hosts via [owfs 1-Wire file system](https://owfs.org/). When using an interface adapter and the owfs, the `mount_dir` option must be configured to correspond to a directory, where owfs device tree has been mounted. On systems where Home Assistant runs in a Docker container. `owfs` cannot escape that environment and hence cannot populate the `mount_dir`. Use the owserver method on these systems instead.
```

File: source/_integrations/onewire.markdown [hunk 10]
```diff
@@ -101,3 +104,42 @@ sensor:
     names:
       GENERATED_ID: FRIENDLY_NAME
 ```
+
+### Entities and attributes
+
+Upon startup of the platform, the 1-wire bus is searched for available 1-wire devices. For each device that this platform handles (see list of supported devices above) the platform adds one sensor for each physical quantity it measures. The name of the sensor is the device ID with the physical quantity it measures appended.
```
Comment (frenck) [inline source/_integrations/onewire.markdown]:
```suggestion
Upon startup of the platform, the 1-wire bus is searched for available 1-wire devices. For each device that this platform handles (see list of supported devices above), the platform adds one sensor for each physical quantity it measures. The name of the sensor is the device ID with the physical quantity it measures appended. 
```

File: source/_integrations/onewire.markdown [hunk 11]
```diff
@@ -101,3 +104,42 @@ sensor:
...
       GENERATED_ID: FRIENDLY_NAME
 ```
+
+### Entities and attributes
+
+Upon startup of the platform, the 1-wire bus is searched for available 1-wire devices. For each device that this platform handles (see list of supported devices above) the platform adds one sensor for each physical quantity it measures. The name of the sensor is the device ID with the physical quantity it measures appended. 
+
+`sensor.28.FF5C68521604_temperature`
+
+Each sensor will present its value in the state variable and for presentation purposes rounded to one decimal for analog values and as integers for counts.
```
Comment (frenck) [inline source/_integrations/onewire.markdown]:
```suggestion
Each sensor will present its value in the state variable and for presentation purposes, rounded to one decimal for analog values and as integers for counts.
```

File: source/_integrations/onewire.markdown [hunk 12]
```diff
@@ -101,3 +104,42 @@ sensor:
...
+
+Each sensor also exposes some additional attirbutes.
```
Comment (frenck) [inline source/_integrations/onewire.markdown]:
```suggestion
Each sensor also exposes some additional attributes.
```

File: source/_integrations/onewire.markdown [hunk 13]
```diff
@@ -101,3 +104,42 @@ sensor:
...
+
+This platform works with devices with multiple sensors which will cause a discontinuity in recorded values. Existing devices will receive a new ID and therefore show up as new devices.
+If you wish to maintain continuity it can be resolved in the database by renaming the old devices to the new names.
+
+Connect to your database using the instructions from [Database section](/docs/backend/database/). Check the names of sensors:
+
+```sql
+SELECT entity_id, COUNT(*) as count FROM states GROUP BY entity_id ORDER BY count DESC LIMIT 10;
+```
+Alter the names of sensors using the following examples:
```
Comment (frenck) [inline source/_integrations/onewire.markdown:137]:
```suggestion

Alter the names of sensors using the following examples:
```

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/onewire.markdown (hunk 3 above)
Comment (MrDadoo) [reply source/_integrations/onewire.markdown]: Thanks allot. I’ll pass my text through spell checking the next time:-)

File: source/_integrations/onewire.markdown (hunk 2 above)
Comment (MrDadoo) [reply source/_integrations/onewire.markdown]: My intentions were internal references. [2] refers to [2] in table. Then they might not be rmoved. 
To note where my explanations were located I used ####Notes becaused it looked good. Its not a heading, so #### might not be good. How about making it bold or italic instead? 
OK to keep reference notations ?

Comment (frenck) [reply source/_integrations/onewire.markdown]: `####` = a heading, it is a `h4`.

Our website does not support Markdown references or footnotes, unfortunately.

**Comment by MrDadoo** [comment]
On the 1-wire bus, sensors might come and go. 
A sensor that has been available might disappear for a while, especially on unstable networks.
I've found last_changed and last_updated attributes (sounds quite similar to me).
But what is best practice to note if a sensor is present or not. On 433 sensors, battery might go low or obstacles may appear in radio transmission and sensors might be away for a while. 
In other systems, this can be shown by the sensor being dimmed or having a ! mark. 
What is the preferred way of implementation in HA?
Also, if new 1-wire devices appear on the network. Is it then good practice to dynamically add them?
For disappeared sensors, removing them is not a good solution in my opinion, but marking them as not available.
I miss a "last_seen" property handled by ha inherently.

**Comment by MrDadoo** [comment]
@frenck and @klaasnicolaas ,in Home Assistant  (and  … Core and ...Supervised), is it really possible to access the GPIO pins on the Raspberry Pi  for example. The doc has descriptions on how to use w1-gpio kernel module. Is that module available on HassOS? (I myself run Home Assistant Supervised on a Linux OS so I cannot check) As HA nowadays run in virtualized python or docker or on HassOS, shouldn't that part be removed from this document? 
We can run the onewire platform using owserver over network or install an add-on that runs some os that can read the I2C or USB busses and expose owfs over network. Even if we in an add-on can run owfs and populate an /mnt/1-wire folder, this onewire integration cannot read that containers file system.
Am I completely off track here, or does the document need a bigger revision?

**Review by frenck — APPROVED** [review]

**Pull request merged**